src/SFA.DAS.Employer.PR.Web/Authentication/EmployerAllRolesAuthorizationHandler.cs

```
@@ -0,0 +1,78 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using SFA.DAS.Employer.PR.Domain.OuterApi.Responses;
+using SFA.DAS.Employer.PR.Web.Infrastructure;
+using System.Text.Json;
+
+namespace SFA.DAS.Employer.PR.Web.Authentication;
+
+
+public class EmployerAllRolesAuthorizationHandler : AuthorizationHandler<EmployerAllRolesRequirement>
+{
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public EmployerAllRolesAuthorizationHandler(IHttpContextAccessor httpContextAccessor)
+    {
+        _httpContextAccessor = httpContextAccessor;
+    }
+
+    protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, EmployerAllRolesRequirement requirement)
+    {
+        if (!IsEmployerAuthorised(context, _httpContextAccessor.HttpContext!.Request.RouteValues, true))
+        {
+            return Task.CompletedTask;
+        }
+
+        context.Succeed(requirement);
+        return Task.CompletedTask;
+    }
+
+    private static bool IsEmployerAuthorised(AuthorizationHandlerContext context, RouteValueDictionary routeValues, bool allowAllUserRoles)
+    {
+        if (!routeValues!.ContainsKey(RouteValueKeys.EncodedAccountId))
+        {
+            return false;
+        }
+
+        var accountIdFromUrl = routeValues[RouteValueKeys.EncodedAccountId]!.ToString()!.ToUpper();
+        var employerAccountClaim = context.User.FindFirst(c => c.Type.Equals(EmployerClaims.AccountsClaimsTypeIdentifier));
+
+        if (employerAccountClaim?.Value == null)
+            return false;
+
+        EmployerUserAccountItem? employerAccount = null;
+        try
+        {
+            var employerAccounts = JsonSerializer.Deserialize<Dictionary<string, EmployerUserAccountItem>>(employerAccountClaim.Value);
+            if (employerAccounts != null && employerAccounts.TryGetValue(accountIdFromUrl, out var value))
+            {
+                employerAccount = value;
+            }
+        }
+        catch
```

I do not like hiding exceptions, if this is unexpected situation, then letting it bubble will highlight the issue, else we will never know why user was not allowed.

---

try catch removed